apps/src/code-studio/pd/components/schoolAutocompleteDropdownWithCustomFields.jsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Autocomplete school dropdown with
+ * custom fields if school not found
+ */
+import React, {PropTypes} from 'react';
+import {FormGroup, Row, Col, ControlLabel, HelpBlock} from 'react-bootstrap';
+import SchoolAutocompleteDropdown from '@cdo/apps/templates/SchoolAutocompleteDropdown';
+import CustomSchoolInfo from './customSchoolInfo';
+import {isZipCode} from '@cdo/apps/util/formatValidation';
+
+
+const VALIDATION_STATE_ERROR = "error";
+const OTHER_SCHOOL_VALUE = "-1";
+
+export default class SchoolAutocompleteDropdownWithCustomFields extends React.Component {
+  static propTypes = {
+    school_id: PropTypes.string,
+    school_info: PropTypes.object,
+    onSchoolInfoChange: PropTypes.func,
+    errors: PropTypes.object
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      school_id: null,
+      school_info: this.props.school_info,
+      errors: this.props.errors || {}
+    };
+  }
+
+  updateOnSchoolInfoChange = (school_info) => {
+    this.setState(school_info);
```

Comment:
Do each of these internal components need state? If the new value is always being passed up the callback chain, then we don't need to store state along the way at all, let alone multiple times.